src/test/ui/impl-trait/equality-recursive.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(conservative_impl_trait, specialization)]
```

Comment:
I see you factored this test out from a separate piece of code that started failing (in a new way) in `src/test/ui/impl-trait/equality.rs` ... do we need to be concerend about this change in behavior? I.e. Does it need a separate issue filed for it, once this PR lands?

Reply:
I split this out since this error happens in another compiler phase compared to the rest. I don't remember the details, but I probably did something which caused the compiler to exit earlier due to errors.